Add API reference for deriveOptions and getDerived method (#31)
ably/sdk-api-reference: descriptions.md

| params?: `Dict<String, String>` ||| TB2c | [Channel Parameters](https://ably.com/docs/realtime/channels/channel-parameters/overview) that configure the behavior of the channel. |
| modes?: [ChannelMode] ||| TB2d | An array of [`ChannelMode`]{@link ChannelMode} objects. |

## class DeriveOptions

> Passes additional properties to a [`RealtimeChannel`]{@link RealtimeChannel} object, such as filter, to create a new derived/synthetic channel.

Reviewer: ```suggestion
Passes properties to a [`RealtimeChannel`]{@link RealtimeChannel} object, such as a filter, to create a derived channel.
```

Suggestion to amend the wording slightly given it's mandatory rather than optional. Also agree with the associated [spec PR](https://github.com/ably/specification/pull/138#discussion_r1184835973) to avoid introducing 'synthetic' as well as 'derived'.

| Method / Property | Parameter | Returns | Spec | Description |
|---|---|---|---|----|

> | filter: `String` ||| RTS5a | JMESPath string expression of filter property to be used to create a Subscription filter derived channel |

Reviewer: ```suggestion
| filter: `String` ||| RTS5a | A filter expression written in JMESPath, used to filter messages received by the derived channel. |
```
